src/App.tsx

```diff
@@ -1,24 +1,248 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState, useEffect } from 'react';
+import classNames from 'classnames';
 import { UserWarning } from './UserWarning';
+import {
+  getTodos,
+  postTodos,
+  deleteTodo,
+  updateTodoStatus,
+  updateTodoTitle,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { Footer } from './components/Footer';
+import { FilterBy } from './types/FilterBy';
+import { getFilteredTodos } from './helpers/getFilteredTodos';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
 
-const USER_ID = 0;
+const USER_ID = 10775;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filterBy, setFilterBy] = useState(FilterBy.ALL);
+  const [isError, setIsError] = useState<string | null>(null);
+  const [inputValue, setInputValue] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [disableInput, setDisableInput] = useState(false);
+  const [loadingTodo, setLoadingTodo] = useState([0]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((loadedTodos: Todo[]) => {
+        setTodos(loadedTodos);
+      })
+      .catch((loadedError: Error) => {
+        setIsError(loadedError?.message ?? 'Error');
+      });
+  }, []);
+
+  useEffect(() => {
+    if (isError) {
+      setTimeout(() => {
+        setIsError(null);
+      }, 3000);
+    }
+  }, [isError]);
+
+  const addTodo = async (todoTitle: string) => {
+    try {
+      const newTodo = {
+        userId: USER_ID,
+        title: todoTitle,
+        completed: false,
+      };
+
+      setTempTodo({
+        id: 0,
+        ...newTodo,
+      });
+
+      const createdTodo = await postTodos(newTodo);
+
+      setTodos(currentTodos => [...currentTodos, createdTodo]);
+    } catch {
+      setIsError('Unable to add a todo');
+    } finally {
+      setTempTodo(null);
+    }
+  };
+
+  const updateTitle = async (
+    id: number,
+    title: string,
+  ) => {
+    try {
+      setLoadingTodo(prevTodosId => [...prevTodosId, id]);
+      const updatingTodoIndex = todos.findIndex(todo => todo.id === id);
```

Comment:
We don't have to look for the index there to update, we can find it directly as an object through `.find`.